src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -65,98 +69,107 @@ func (suite *DeltaTreeUnitSuite) TestNewNodeyMcNodeFace() {
 // note that this test is focused on the SetFolder function,
 // and intentionally does not verify the resulting node tree
 func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_SetFolder() {
+	d := drive()
+
 	table := []struct {
 		tname     string
 		tree      func(t *testing.T, d *deltaDrive) *folderyMcFolderFace
-		parentID  string
-		id        string
-		name      string
-		isPackage bool
+		folder    func() *custom.DriveItem
 		expectErr assert.ErrorAssertionFunc
 	}{
 		{
-			tname:     "add root",
-			tree:      newTree,
-			id:        rootID,
-			name:      rootName,
-			isPackage: true,
+			tname: "add root",
+			tree:  newTree,
+			folder: func() *custom.DriveItem {
+				return custom.ToCustomDriveItem(rootFolder())
+			},
 			expectErr: assert.NoError,
 		},
 		{
-			tname:     "root already exists",
-			tree:      treeWithRoot,
-			id:        rootID,
-			name:      rootName,
+			tname: "root already exists",
+			tree:  treeWithRoot,
+			folder: func() *custom.DriveItem {
+				return custom.ToCustomDriveItem(rootFolder())
+			},
 			expectErr: assert.NoError,
 		},
 		{
-			tname:     "add folder",
-			tree:      treeWithRoot,
-			parentID:  rootID,
-			id:        folderID(),
-			name:      folderName(),
+			tname: "add folder",
+			tree:  treeWithRoot,
+			folder: func() *custom.DriveItem {
+				return custom.ToCustomDriveItem(d.folderAtRoot())
+			},
 			expectErr: assert.NoError,
 		},
 		{
-			tname:     "add package",
-			tree:      treeWithRoot,
-			parentID:  rootID,
-			id:        folderID(),
-			name:      folderName(),
-			isPackage: true,
+			tname: "add package",
+			tree:  treeWithRoot,
+			folder: func() *custom.DriveItem {
+				return custom.ToCustomDriveItem(
```

Comment:
nit: since I think there's a helper can this be `custom.ToCustomDriveItem(d.pkgAtRoot())`?

Reply:
right you are.